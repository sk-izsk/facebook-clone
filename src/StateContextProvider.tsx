import React, { createContext, useContext, useReducer } from 'react';

export interface StateContextProviderProps {
  initialState?: any;
  reducer?: any;
}

const StateContext = createContext(undefined);

const StateContextProvider: React.FC<StateContextProviderProps> = ({ initialState, reducer, children }) => {
  return <StateContext.Provider value={useReducer(reducer, initialState) as any}>{children}</StateContext.Provider>;
};

const useStateValue: any = () => useContext(StateContext);

export { StateContextProvider, StateContext, useStateValue };
