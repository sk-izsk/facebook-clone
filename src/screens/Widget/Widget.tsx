import { Box, makeStyles } from '@material-ui/core';
import React from 'react';
import { CustomTheme } from '../../theme/muiTheme';

export interface WidgetProps {}

const useStyles = makeStyles((theme: CustomTheme) => ({
  widgets: {},
  plugin: {
    border: 'none',
    overflow: 'hidden',
  },
}));

const Widget: React.FC<WidgetProps> = () => {
  const classes = useStyles();
  return (
    <Box className={classes.widgets}>
      <iframe
        src='https://www.facebook.com/plugins/page.php?href=https%3A%2F%2Fwww.facebook.com%2FFac2Pedia-1818827908414968%2F&tabs=timeline&width=340&height=500&small_header=false&adapt_container_width=true&hide_cover=false&show_facepile=true&appId'
        width='340'
        height='100%'
        scrolling='no'
        frameBorder='0'
        allowTransparency={true}
        allow='encrypted-media'
        title='fb-plugin'
        className={classes.plugin}
      ></iframe>
    </Box>
  );
};

export { Widget };
