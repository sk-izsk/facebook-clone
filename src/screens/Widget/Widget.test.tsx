import { render } from '@testing-library/react';
import React from 'react';
import { StateContextProvider } from '../../StateContextProvider';
import { Widget } from './Widget';

const mockUser = {
  image: 'fake image url',
  message: 'fake message',
  profilePic: 'fake profile pic',
  timestamp: 'fake time stamp',
  username: 'fake username',
};

test('Widget renders correctly', () => {
  const { asFragment } = render(
    <StateContextProvider initialState={{ user: mockUser }}>
      <Widget />
    </StateContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
