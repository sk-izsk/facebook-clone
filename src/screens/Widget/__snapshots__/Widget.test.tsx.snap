// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-3 makeStyles-widgets-1"
  >
    <iframe
      allow="encrypted-media"
      class="makeStyles-plugin-2"
      frameborder="0"
      height="100%"
      scrolling="no"
      src="https://www.facebook.com/plugins/page.php?href=https%3A%2F%2Fwww.facebook.com%2FFac2Pedia-1818827908414968%2F&tabs=timeline&width=340&height=500&small_header=false&adapt_container_width=true&hide_cover=false&show_facepile=true&appId"
      title="fb-plugin"
      width="340"
    />
  </div>
</DocumentFragment>
`;
